lib/ThreeElementVectorLoweringPass.cpp

```diff
@@ -804,13 +837,17 @@ clspv::ThreeElementVectorLoweringPass::visitUnaryOperator(UnaryOperator &I) {
 }
 
 bool clspv::ThreeElementVectorLoweringPass::handlingRequired(User &U) {
-  if (getEquivalentType(U.getType()) != nullptr) {
+  if (getEquivalentType(clspv::InferType(&U, U.getContext(), &type_cache_)) !=
+      nullptr) {
     return true;
   }
 
   for (auto &Operand : U.operands()) {
     auto *OperandTy = Operand.get()->getType();
-    if (getEquivalentType(OperandTy) != nullptr) {
+    if (OperandTy->isOpaquePointerTy()) {
+      OperandTy = clspv::InferType(Operand, U.getContext(), &type_cache_);
+    }
+    if (OperandTy != nullptr && getEquivalentType(OperandTy) != nullptr) {
```

Comment:
`OperandTy != nullptr` is a guard as InferType could return nullptr like when it cannot guess the opaque pointer type.